AK/Optional.h

```diff
@@ -261,6 +288,25 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> : public Optiona
         return *this;
     }
 
+    template<class U = T>
+    requires(
+        !OneOf<RemoveCVReference<U>, Optional, OptionalNone>
+        && !(IsSame<U, T> && IsScalar<U>))
```

Comment:
Maybe add a not about the `IsScalar` like I did in serenity, as that may seem unrelated, but makes the `= {}` work as intended

Reply:
I've added a comment :+1: